File: development/NET/samples/server/Session/VB/MyDaAddressSpaceRoot.vb
Imports Softing.OPCToolbox
Imports Softing.OPCToolbox.Server

Namespace Session
	Public Class MyDaAddressSpaceRoot
		Inherits DaAddressSpaceRoot
#Region "Public Methods"
		'--------------------

		Public Overloads Overrides Function QueryAddressSpaceElementData(ByVal elementId As String, ByRef anAddressSpaceElement As AddressSpaceElement) As Integer
			'  TODO: add string based address space validations
			anAddressSpaceElement = Nothing
			Return EnumResultCode.E_NOTIMPL
		End Function

		'--
#End Region

	End Class
End Namespace

File: development/NET/samples/server/Session/CS/MyDaAddressSpaceRoot.cs
using Softing.OPCToolbox;
using Softing.OPCToolbox.Server;

namespace Session
{
	public class MyDaAddressSpaceRoot : DaAddressSpaceRoot
	{
		#region Public Methods
		//--------------------

		public override int QueryAddressSpaceElementData(
			string elementId,
			out AddressSpaceElement anAddressSpaceElement)
		{
			//	TODO: add string based address space validations
			anAddressSpaceElement = null;
			return (int)EnumResultCode.E_NOTIMPL;
		}   //	end QueryAddressSpaceElementData

		//--
		#endregion

	}   //	end MyDaAddressSpaceRoot
}   //	end namespace
